### ISSUE TITLE
Link POIs to events in event form

### ISSUE THREAD

**Issue by timoludwig** [Integreat/cms-django#247]
Make it possible to add existing POIs as location for events or create a new POI if an event with a new location is added.
Ideally, make it possible to search for existing POIs instead of having a select tag with dozens of options.

**Comment by timoludwig** [comment]
@florianhofhammer I think the fields "slug" and "description" can be left out in the poi form on the event view. I we still have too many fields and the event form is too complex with an integrated poi form, we could think about a popup with the poi form in it (similar wo what django-admin is doing)... if both of these approaches are too cumbersome, we could also leave out the poi form entirely and just show a message that pois have to be created via the poi form first before they can be selected as venue for an event... or what do you think?

**Comment by florianhofhammer** [comment]
I think it is not a good idea to make a poi completely editable in the event form, as all the needed fields from the poi form would just unnecessarily overload the event form.
I think the following approach would make sense:
* Use the fields as we currently do (poi title, address, city, country)
* Make only the title field editable; the other ones are just read-only and showing additional information
* Have "live search" in the title field: while you're typing something, a list of pois whose title matches the current input shows up below the field ("matching" can mean the beginning of the title matches, the input appears somewhere in the title, the input appears somewhere in the description, etc. I think we should talk about this). In this list, the first entry is **always** "Create new entry "\<current input string\>"". Selecting any other option than that one just assigns the selected poi as the poi to the event. Selecting this first option opens up a complete poi form and allows to create a new poi.

The disadvantage of this approach is that users always have to open the poi form in order to create/edit pois. The huge advantage, however, is that we don't overload the event form. I think that might be beneficial for the usability in the end, even though one or two additional mouse clicks might be necessary to achieve the desired result.

**Comment by timoludwig** [comment]
Yes, this sounds like an extremely good idea! From a UI/UX-perspective, this would be a nearly ideal approach. The downside is that it's maybe a bit difficult to implement, but in the long-term it will probably pay off. One question remains open to me: What happens, if the user already made changes to the event form and then selects "Create new entry" as poi? Will the user be redirected to the poi form and the changes discarded, or will the field set to "this event has no location", saved as draft and then the user is redirected to the poi form? Or can we manage to show a pop-up/overlay poi form, which closes after saving and returns to the original event form with the newly created poi selected as location?

**Comment by florianhofhammer** [comment]
I would prefer either the approach where the event is saved as draft without a location or an approach where a popup window notifies the user of unsaved changes and provides the option to either save as draft and continue to the poi form, save and publish and continue to the poi form or cancel. I personally think a popup or overlay with the complete poi form would make the implementation unnecessarily complex. Also, if this popup or overlay was implemented as a new version of the poi form, we would have to maintain two different templates/forms in the future. If we reused the existing poi form, we would depend on this poi form in two different places (the original form and the event popup/overlay). This would also negatively impact maintainability in my eyes.
